src/llmq/quorums_dkgsessionhandler.cpp

```diff
@@ -120,24 +120,11 @@ void CDKGSessionHandler::UpdatedBlockTip(const CBlockIndex* pindexNew, const CBl
     quorumHeight = pindexQuorum->nHeight;
     quorumHash = pindexQuorum->GetBlockHash();
 
-    QuorumPhase newPhase = phase;
-    if (quorumStageInt == 0) {
-        newPhase = QuorumPhase_Initialized;
-        quorumDKGDebugManager->ResetLocalSessionStatus(params.type, quorumHash, quorumHeight);
-    } else if (quorumStageInt == params.dkgPhaseBlocks * 1) {
-        newPhase = QuorumPhase_Contribute;
-    } else if (quorumStageInt == params.dkgPhaseBlocks * 2) {
-        newPhase = QuorumPhase_Complain;
-    } else if (quorumStageInt == params.dkgPhaseBlocks * 3) {
-        newPhase = QuorumPhase_Justify;
-    } else if (quorumStageInt == params.dkgPhaseBlocks * 4) {
-        newPhase = QuorumPhase_Commit;
-    } else if (quorumStageInt == params.dkgPhaseBlocks * 5) {
-        newPhase = QuorumPhase_Finalize;
-    } else if (quorumStageInt == params.dkgPhaseBlocks * 6) {
-        newPhase = QuorumPhase_Idle;
+    bool fNewPhase = quorumStageInt % params.dkgPhaseBlocks == 0;
```

Comment:
Please add parentheses around `quorumStageInt % params.dkgPhaseBlocks`...Even after years I'm unable to remember the operator precedence with `%` (not an issue with `==`, but gets funny when mixed with `+` and `-` if I remember correct).

```suggestion
    bool fNewPhase = (quorumStageInt % params.dkgPhaseBlocks) == 0;
```